mne/coreg.py

```diff
@@ -1242,3 +1245,613 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+
+        self._dig = _DigSource(info)
+        self._mri = _MRIHeadWithFiducialsModel(subjects_dir=subjects_dir,
+                                               subject=subject)
+        self.reset()
+        self._nearest_calc = self._nearest_calc_default()
+
+    def _update_params(self, rot=None, tra=None, sca=None):
+        rot_changed = False
+        if self._rot_trans is None or rotation is not None:
+            rot_changed = True
+            self._last_rotation = self._rotation
+            self._rotation = rot
+            self._rot_trans = rotation(*rot).T
+        tra_changed = False
+        if rot_changed or tra is not None:
+            tra_changed = True
+            self._last_translation = self._translation
+            self._translation = tra
+            self._head_mri_t = self._rot_trans.copy()
+            self._head_mri_t[:3, 3] = \
+                -np.dot(self._head_mri_t[:3, :3], tra)
+            self._mri_head_t = self._rot_trans.copy()
+            self._mri_head_t[:3, 3] = np.array(tra)
+        if tra_changed or sca is not None:
+            sca = self._scale if sca is None else sca
+            self._last_scale = self._scale
+            self._scale = sca
+            if self._coord_frame == 'head':
+                self._mri_trans = self._mri_head_t.copy()
+            else:
+                self._mri_trans = np.eye(4)
+            self._mri_trans[:, :3] *= sca
+
+    def set_rotation(self, rot):
+        """Set the rotation parameter.
+
+        Parameters
+        ----------
+        rot : array, shape (3,)
+            The rotation parameter.
+        """
+        self._update_params(rot=rot)
+
+    def set_translation(self, tra):
+        """Set the translation parameter.
+
+        Parameters
+        ----------
+        tra : array, shape (3,)
+            The translation parameter.
+        """
+        self._update_params(tra=tra)
+
+    def set_scale(self, sca):
+        """Set the scale parameter.
+
+        Parameters
+        ----------
+        sca : array, shape (3,)
+            The scale parameter.
+        """
+        self._update_params(sca=sca)
+
+    @property
+    def _parameters(self):
+        return np.concatenate((self._rotation, self._translation, self._scale))
+
+    @property
+    def _last_parameters(self):
+        return np.concatenate((self._last_rotation,
+                               self._last_translation, self._last_scale))
+
+    @property
+    def _changes(self):
+        move = np.linalg.norm(self._last_translation - self._translation) * 1e3
+        angle = np.rad2deg(_angle_between_quats(
+            rot_to_quat(rotation(*self._rotation)[:3, :3]),
+            rot_to_quat(rotation(*self._last_rotation)[:3, :3])))
+        percs = 100 * (self._scale - self._last_scale) / self._last_scale
+        return move, angle, percs
+
+    @property
+    def _transformed_dig_hpi(self):
+        return apply_trans(self._hsp_trans, self._dig.hpi_points)
```

Comment:
Let's not `@property` any of these, instead update all of them as private attributes in `_update_params` as needed